File: OneCodeTeam/How-to-use-Windows-Azure-ACS-to-authenticate-in-25335/[VB]-How to use Windows Azure ACS to authenticate in WPF application/VB/VBAzureACSAuthInWPF/RESTfulWCFService/IUserService.vb
Imports System.ServiceModel.Web

<ServiceContract> _
Interface IUserService
    <WebGet(UriTemplate:="/users", ResponseFormat:=WebMessageFormat.Xml)> _
    <OperationContract> _
    Function GetAllUsers() As Users
    <WebInvoke(UriTemplate:="/users", Method:="POST", RequestFormat:=WebMessageFormat.Xml, ResponseFormat:=WebMessageFormat.Xml)> _
    <OperationContract> _
    Function AddNewUser(u As User) As User

    <WebGet(UriTemplate:="/users/{user_id}", ResponseFormat:=WebMessageFormat.Xml)> _
    <OperationContract> _
    Function GetUser(user_id As String) As User
End Interface

File: OneCodeTeam/How-to-use-Windows-Azure-ACS-to-authenticate-in-31212/[C#]-How to use Windows Azure ACS to authenticate in WPF application/C#/CSAzureACSAuthInWPF/RESTfulWCFService/IUserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ServiceModel;
using System.Text;
using System.Threading.Tasks;
using System.ServiceModel.Web;

namespace RESTfulWCFService
{
    [ServiceContract]
    interface IUserService
    {
        [WebGet(UriTemplate = "/users",
                ResponseFormat = WebMessageFormat.Xml)]
        [OperationContract]
        Users GetAllUsers();
        [WebInvoke(UriTemplate = "/users",
                   Method = "POST",
                   RequestFormat = WebMessageFormat.Xml,
                   ResponseFormat = WebMessageFormat.Xml)]
        [OperationContract]
        User AddNewUser(User u);

        [WebGet(UriTemplate = "/users/{user_id}",
                ResponseFormat = WebMessageFormat.Xml)]
        [OperationContract]
        User GetUser(string user_id);
    } 
}
